c/flake.nix

```diff
@@ -0,0 +1,42 @@
+{
+  description = "C Template";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs";
+    systems.url = "github:nix-systems/x86_64-linux";
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.systems.follows = "systems";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, systems }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        pname = "hello-world"; #package name
+        version = "0.0.1";
+        src = ./.;
+        buildInputs = with pkgs; [
+          gcc
```

Comment:
Unfortunately this is not how you properly set gcc as the default compiler.
There is already a C compiler included in stdenv.mkDerivation and to change you can use `overrideCC`: https://nixos.wiki/wiki/C#Use_a_different_compiler_version

Also just a c compiler probably is not enough for most real-world projects.
Instead of languages, how about making them build system specific. i.e. one for cmake, one for autotools and one for meson?

Reply:
Good points. I'm not a c dev so I'm not that familiar with the ecosystem. This is enough for my purposes of having a hello world run so that I can practice.

Maybe it's worth having this for both mac and linux + the system-specific project templates separately.

Reply:
@Mic92 Can you provide a set of example projects for cmake, one for autotools and one for meson so that I can get them working please

Reply:
Here is a template that we use: https://github.com/TUM-DSE/doctor-cluster-config/blob/50e097983b016bc7c9704ecd1c6cc9ba162f49d5/templates/project-template/flake.nix#L20

Potentially one could also just have one template for both c/c++ but just comment out common build systems:

For cmake add `cmake` in nativeBuildInputs
For meson add `ninja`/`meson` in nativeBuildInputs
For autotools add `autoconf-archive` and `autoreconfHook`
Then I would just add `pkg-config` unconditionally to nativeBuildInputs as a reasonable default.

Reply:
Looks good to me